front/app/utils/cl-router/useQuery.ts

```diff
@@ -0,0 +1,8 @@
+import { useMemo } from 'react';
+import { useLocation } from 'react-router-dom';
+
+export default function useQuery() {
```

Comment:
Can replace all the places where we use `new URLSearchParams`, `parse` from the `qs` package and use search from useLocation.